spring-integration-core/src/main/java/org/springframework/integration/util/MessagingMethodInvokerHelper.java

```diff
@@ -267,8 +268,22 @@ else if (targetMethod instanceof String) {
 	private void prepareEvaluationContext() {
 		StandardEvaluationContext context = getEvaluationContext(false);
 		Class<?> targetType = AopUtils.getTargetClass(this.targetObject);
+
+		ReflectiveMethodResolver declaredMethodResolver = new ReflectiveMethodResolver() {
+
+			@Override
+			protected Method[] getMethods(Class<?> type) {
+				return Stream.of(type.getMethods(), type.getDeclaredMethods())
+						.flatMap(Stream::of)
+						.toArray(Method[]::new);
```

Comment:
I presume you realized we eliminated super class methods with the previous commit. Perhaps we need a test case for that.

Also, I think we need to weed-out duplicates here to shorten the iteration.